File: LogicalThinkingPatterns/InvertedTrianglePattern/InvertedTriangle.ts
class InvertedTriangle {
  printPattern(n: number): void {
    for (let i = n; i > 0; i--) {
      let pattern: string = "";
      for (let j = 0; j < i; j++) {
        pattern += "* ";
      }
      console.log(pattern);
    }
  }
}

const triangle = new InvertedTriangle();
triangle.printPattern(5);

File: LogicalThinkingPatterns/InvertedTrianglePattern/InvertedTriangle.py
class InvertedTriangle:
    def printPattern(self,n):
        for i in range(n, 0, -1):
            pattern = ""
            for j in range(i):
                pattern += "* "
            print(pattern)
    
    def run(self):
        n = int(input("Enter number of rows: "))
        self.printPattern(n)

triangle = InvertedTriangle()
triangle.run()